prometheus/src/snapshot_metrics.rs

```diff
@@ -0,0 +1,65 @@
+use crate::utils::{write_metric, Metric, MetricFamily};
+use solana_runtime::snapshot_archive_info::{SnapshotArchiveInfo, SnapshotArchiveInfoGetter};
+use solana_runtime::snapshot_config::SnapshotConfig;
+use solana_runtime::snapshot_utils;
+use solana_sdk::clock::Slot;
+use std::io;
+
+pub fn write_snapshot_metrics<W: io::Write>(
+    snapshot_config: &Option<SnapshotConfig>,
+    out: &mut W,
+) -> io::Result<()> {
+    let snapshot_config = match snapshot_config.as_ref() {
+        Some(config) => config,
+        None => return Ok(()),
+    };
+
+    let full_snapshot_info = match get_full_snapshot_info(snapshot_config) {
+        Some(info) => info,
+        None => return Ok(()),
+    };
+    write_metric(
+        out,
+        &MetricFamily {
+            name: "solana_full_snapshot_info",
```

Comment:
```suggestion
            name: "solana_snapshot_last_full_snapshot_slot",
```

If we group the parts of the name from less specific to more specific, the metrics become a bit more discoverable in the the Prometheus autocomplete. E.g. you type `solana_`, and then you see there’s `solana_snapshot_`, and then you see there are full snapshots and incremental snapshots. If it’s `solana_full_snapshot`, it’s not obvious that there is also an incremental snapshot among the many metrics.

Also, the metric indicates a slot, so we should call it like that.